### ISSUE TITLE
Including @property in generated entities

### ISSUE THREAD

**Issue by davincho** [pomm-project/Cli#9]
Currently, entities are generated without any property annotation, so code completion in editors is not supported.

One possibility (in my opinion the cleanest one) would be, to change the entity template like this:

```php
namespace {:namespace:};

use {:flexible_container:};

/**
 * {:entity:}
 *
 * Flexible entity for relation
 * {:schema:}.{:relation:}
 *
 * @see FlexibleEntity
 * 
 * @property String name
 * @property Integer ....
 */
class {:entity:} extends {:flexible_container_class:}
{
}
```

Now ```$entity->name``` is available as a code completion suggestion.

**Comment by tlode** [comment]
I've looked into this. It's not that hard to add the `@property` definitions for all column names. But I didn't find a map to resolve the appropriate php type from a pg type.

One solution could be (and would be probably sufficient) to just add the property names without a type definition.

```php
/**
 * {:entity:}
 *
 * Flexible entity for relation
 * {:schema:}.{:relation:}
 *
 * @see FlexibleEntity
 * 
 * @property $chu
 * @property $pika
 */
class {:entity:} extends {:flexible_container_class:}
```

But another way could be to extend the `ConverterInterface` by a method to retrieve the appropriate PHP type.

```php
   /**
     * toPhpType
     *
     * Returns the PHP type representation
     *
     * @access public
     * @param string $type
     * @return string PHP type name
     */
    public function toPhpType($type);
```

E.g. for `PgNumber` this could be implemented like this

```php
    /**
     * @see ConverterInterface
     */
    public function toPhpType($type)
    {
        switch($type) {
            case 'numeric':
            case 'float4':
            case 'float8':
                return 'float';
        }

        return 'int';
    }
```
For a simpler type like `PgBoolean` by just returning `bool`. And for a complex type by returning a class name, e.g. \DateTime 

I did a rough first approach which is not tested yet for complex types, but it could be a starting point. You can see it here
https://github.com/tlode/Foundation/commit/9b85e2c54e21a75919fcdd34c0c9ef8a5111e7de
and here (entity generation)
https://github.com/tlode/ModelManager/compare/master...tlode:doc-properties

**Comment by chanmix51** [comment]
It is important to keep in mind that theses entities are flexible hence, they do not have a fix set of attributes ( since it depends on the projection ). Even though we could annotate the class after the associated `RowStructure`, since the entity class is generated only once, it would not be up to date after a single schema change.

**Comment by tlode** [comment]
Yes, sure it is flexible, but on the other hand, Pomm is also quite strict about this. Think of trying to get a field from an entity which wasn't included in the projection. This would instantly throw an exception.

I am not saying that this is bad or good. Most of the time I need and love this flexibility. Sometimes I rather had an object with real and defined properties, not having to keep in mind which properties are actually represented in this entity after a query. And in those moments I thought of this as a good idea, to annotate at least the `RowStructure`.

Never the less, if there is no other way or benefit from extending the converter interface, then this was just some exercise, getting to know Pomm a little bit more.

Leaves me wondering why the enhancement label then? :-)

**Comment by chanmix51** [comment]
To keep good ideas to come ;)
